src/Model/Item.php

```diff
@@ -2783,6 +2786,27 @@ private static function tagDeliver($uid, $item_id)
 		return false;
 	}
 
+	/**
+	 * Automatically reshare the item if the "remote_self" option is selected
+	 *
+	 * @param array $item
+	 * @return void
+	 */
+	private static function autoReshare(array $item)
+	{
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			return;
+		}
+
+		if (!DBA::exists('contact', ['id' => $item['contact-id'], 'remote_self' => Contact::MIRROR_NATIVE_RESHARE])) {
+			return;
+		}
+
+		Logger::info('Automatically reshare item', ['uid' => $item['uid'], 'id' => $item['id'], 'guid' => $item['guid'], 'uri-id' => $item['uri-id']]);
+
+		Item::performActivity($item['id'], 'announce', $item['uid']);
```

Comment:
Where is the information about the resharer here?

Reply:
The resharer is the given user.